src/components/RegistrationForm/RegistrationForm.tsx

```diff
@@ -0,0 +1,198 @@
+import { useState, FormEvent, ChangeEvent, useMemo } from 'react';
+import { Link } from 'react-router-dom';
+
+import { FormInput } from '../FormInput/FormInput';
+import style from './RegistrationForm.module.scss';
+
+type FormData = {
+  firstName: string;
+  lastName: string;
+  email: string;
+  password: string;
+  confirmPassword: string;
+};
+
+type TouchedFields = {
+  firstName: boolean;
+  lastName: boolean;
+  email: boolean;
+  password: boolean;
+  confirmPassword: boolean;
+};
+
+const INITIAL_FORM_DATA: FormData = {
+  firstName: '',
+  lastName: '',
+  email: '',
+  password: '',
+  confirmPassword: '',
+};
+
+const INITIAL_TOUCHED: TouchedFields = {
+  firstName: false,
+  lastName: false,
+  email: false,
+  password: false,
+  confirmPassword: false,
+};
+
+const ERROR_MESSAGES = {
+  firstName: 'First name cannot be empty or contain numbers',
+  lastName: 'Last name cannot be empty or contain numbers',
+  email: 'Please enter a valid email',
+  password:
+    'Password must have at least 8 characters, 1 uppercase letter and 1 number',
+  passwordMatch: "Passwords don't match",
+};
+
+const validators = {
+  name: (value: string): boolean => /^[^0-9]+$/.test(value) && value.length > 0,
+  email: (value: string): boolean => /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value),
+  password: (value: string): boolean =>
+    /^(?=.*[A-Z])(?=.*\d).{8,}$/.test(value),
+};
```

Comment:
I am not overly happy with generating regex, which only chatgpt can understand. On top of it, I am not sure why we once validate the length via the regex and for name we do it via normal "javascript".

I suggest we drop both password and name validations via regex and we write it BY HAND (NO AI), via classic JS. **Keep it simple.**